weave/flow/dataset.py

```diff
@@ -64,3 +64,21 @@ def convert_to_table(cls, rows: Any) -> weave.Table:
                     "Attempted to construct a Dataset row with an empty dict."
                 )
         return rows
+
+    def __eq__(self, other: Any) -> bool:
+        return self.rows == other
+
+    def __iadd__(self, rows: list[dict]) -> None:
+        if not all(isinstance(row, dict) for row in rows):
+            raise ValueError("Can only append dicts to Dataset")
+        for row in rows:
+            self.rows.append(row)
+        return self
```

Comment:
`__iadd__` support for `ds += [...]`

This is experimental, please see the other `__iadd__` on `WeaveObject` to see if we want to support this pattern.

From a user perspective it does feel more ergonomic, but these classes can quickly get branchy with a bunch of different cases like we have in `make_trace_object`.  I find the branching harder to read, but maybe it's ok?